packages/next/src/client/components/is-hydration-error.ts

```diff
@@ -1,12 +1,13 @@
 import isError from '../../lib/is-error'
 
 const hydrationErrorRegex =
-  /hydration failed|while hydrating|content does not match|did not match|HTML didn't match/i
+  /hydration failed|while hydrating|content does not match|did not match|HTML didn't match|text didn't match/i
 
 const reactUnifiedMismatchWarning = `Hydration failed because the server rendered HTML didn't match the client. As a result this tree will be regenerated on the client. This can happen if a SSR-ed Client Component used:`
 
 const reactHydrationStartMessages = [
   reactUnifiedMismatchWarning,
+  `Hydration failed because the server rendered text didn't match the client. As a result this tree will be regenerated on the client. This can happen if a SSR-ed Client Component used:`,
```

Comment:
https://github.com/facebook/react/pull/32763

The default message should still read "HTML". Though this does highlight that we currently drop the original message. So if you do have a text mismatch, we still unintentionally say "rendered HTML" instead of "rendered text". We'll see if this gets fixed during the upcoming refactors. Not too urgent to fix.